tools/generate_json_map_class.js

```diff
@@ -751,6 +995,177 @@ using cfd::core::JsonVector;
+    path = path.substr(1);
+  }
+  while (path.indexOf('../') >= 0) {
+    path = path.replace('../', '');
+  }
+  while (path.indexOf('./') >= 0) {
+    path = path.replace('./', '');
+  }
+  while (path.indexOf('//') >= 0) {
+    path = path.replace('//', '/');
+  }
+  while (path.indexOf('external/') >= 0) {
+    path = path.replace('external/', '');
+  }
+  while (path.indexOf('/') >= 0) {
+    path = path.replace('/', '_');
+  }
+  while (path.indexOf('.') >= 0) {
+    path = path.replace('.', '_');
+  }
+  while (path.indexOf('-') >= 0) {
+    path = path.replace('-', '_');
+  }
+  const def_name = path.toUpperCase();
+  const include_header = (json_setting.common_header) ? `#include "${json_setting.common_header}"\n` : '';
+  const include_header2 = (append_header_name.length > 0) ? `#include "${append_header_name}"\n` : '';
+
+  // header
+  const header_file_header = `// Copyright 2019 CryptoGarage
+/**
+ * @file ${filename}
+ *
+ * @brief JSON mapping file. (auto generate)
+ */
+#ifndef ${def_name}
+#define ${def_name}
+
+#include <set>
+#include <string>
+#include <vector>
+
+#include "cfdcore/cfdcore_json_mapping_base.h"
+${include_header}
+${include_header2}`;
+
+  result.push(header_file_header);
+
+  if (isArray(namespace)) {
+    for (let idx = 0; idx < namespace.length; ++idx) {
+      result.push(`namespace ${namespace[idx]} {`);
+    }
+  } else {
+    result.push(`namespace ${namespace} {`);
+  }
+
+  const header_file_header2 = `
+using cfd::core::JsonClassBase;
+using cfd::core::JsonObjectVector;
+using cfd::core::JsonValueVector;
+using cfd::core::JsonVector;
+// clang-format off
+// @formatter:off\
+`;
+
+  const header_file_footer = `
+// @formatter:on
+// clang-format on
+`;
+  const header_file_footer2 = `
+#endif  // ${def_name}
+`;
+  result.push(header_file_header2);
+
+  if (class_list) {
+    for (const data in class_list) {
+      result.push(class_list[data]);
+    }
+  }
+
+  result.push(header_file_footer);
+  if (isArray(namespace)) {
+    for (let idx = namespace.length - 1; idx >= 0; --idx) {
+      result.push(`}  // namespace ${namespace[idx]}`);
+    }
+  } else {
+    result.push(`}  // namespace ${namespace}`);
+  }
+  result.push(header_file_footer2);
+  return result.join('\n');
+};
+
+// ----------------------------------------------------------------------------
+// generate class header function
+// ----------------------------------------------------------------------------
+const generateClassHeader = (req, res, json_setting) => {
+  const result = [];
+  const processed_list = [];
+  const export_define = (json_setting.export) ? `${json_setting.export} ` : '';
+
+  // header
+  if (req || res) {
+    const list = [req, res];
+    for (const data in list) {
+      if (!data) continue;
+      // sort by generate class
+      // Child element in order
+      const map_list = [];
+      getChildClasses(list[data], map_list);
+      // console.log('map_list = ${map_list}`)
```

Comment:
You have a lot of commented out `console.log`. I think either delete them or uncomment them and print out a meaningful message (it's an internal tool so I think it's fine that it prints out some information).

Reply:
Thanks! 🙏 
Changed to log debug information when `mode=debug` argument is specified.